### DISCUSSION TITLE
Newly Created Menu items not deleting

### DISCUSSION THREAD

**Discussion by elliewalkersm** [nss-evening-cohort-14/evening-client-side#196] [Help Me!] [answered]
Your Name: Ellie Walker

Have you searched the other help tickets?
Yes I have.

Describe your issue
My delete function is working on the menu items from my mock data, however, it is not working on the menu items created using my form. 

List any error messages that appear
No error message.

What have you googled?
https://stackoverflow.com/questions/61688160/delete-function-not-working-in-newly-created-item
https://laracasts.com/discuss/channels/laravel/delete-function-not-working
https://stackoverflow.com/questions/33239464/javascript-delete-object-property-not-working

What have you tried? Tell us everything you have tried.
-I tried adding a console.warn in place of showLoginMenuItems in my delete eventListener. The mock data returns the array minus the one deleted, the new data returns the full array without deleting the one clicked.
-I tried changing the is split from '--' to '^^' as this fixed an issue for one of my teammates.
-I reached out to my teammates to look over the code.

Links to my repo

menuData: https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/ejw-create-menu/src/javascripts/helpers/data/menuData.js

domEvents: https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/ejw-create-menu/src/javascripts/events/domEvents.js

createMenuItemForm: https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/ejw-create-menu/src/javascripts/components/forms/createMenuItemForm.js

**Comment by caseywalker** [comment, marked as answer]
Using 'firebasekey' in your sample data and domevents. New objects are assigned 'firebaseKey' so the domevent returns null on newly created objects. 

Mixing firebasekey and firebaseKey. 

Solution: refactor the createMenuItem Promise to call it 'firebasekey' or... refactor sample data, domEvents, and buttons to firebaseKey.

**Reply by caseywalker** [reply]
Sample data: https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/main/json_sample_data/menu.json#L8

**Reply by caseywalker** [reply]
delete button refers to 'firebasekey' https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/ejw-create-menu/src/javascripts/components/menu/menu.js#L34

**Reply by caseywalker** [reply]
Post/Patch 'create' API call patches with 'firebaseKey' 
https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/ejw-create-menu/src/javascripts/helpers/data/menuData.js#L26

**Reply by elliewalkersm** [reply]
That was it, thanks!
